main.go

```diff
@@ -222,13 +223,30 @@ func main() {
 	result, platformsDetected := scanner.GenerateScanResult(searchDir, hasSSHKey)
 
 	// Upload results
-	if resultClient != nil {
-		log.TInfof("Submitting results...")
-		if err := resultClient.uploadResults(result); err != nil {
-			failf("Could not send back results: %s", err)
+	var resultBytes []byte
+	shouldStoreResult := resultClient != nil || strings.HasPrefix(cfg.ResultSubmitURL, "path::")
+	if shouldStoreResult {
+		resultBytes, err = json.MarshalIndent(result, "", "\t")
+		if err != nil {
+			failf("failed to marshal results: %v", err)
 		}
 
-		log.TDonef("Submitted.")
+		if resultClient != nil || !strings.HasPrefix(cfg.ResultSubmitURL, "path::") {
+			log.TInfof("Submitting results...")
+			if err := resultClient.uploadResults(resultBytes); err != nil {
+				failf("Could not send results: %s", err)
+			}
+
+			log.TDonef("Results submitted.")
+		} else if strings.HasPrefix(cfg.ResultSubmitURL, "path::") {
```

Comment:
It was not clear at first what this new local save feature means. Can you add some explanation that this is not used in prod and it is mostly for local testing with the sample workflow?

Reply:
Updated in https://github.com/bitrise-steplib/steps-project-scanner/pull/148/commits/bdcb0dae4a42cf642571062fc889b7db71efd8b4